### PR TITLE
allow s2s bidders call pbs without need of a client adapter file

### PR DESCRIPTION
**Pull request by jsnellbaker**
## Type of change
- [x] Feature

## Description of change
To add support for the request noted in #2659 

This will allow a pbs bidder to make a request through prebid.js without needing a corresponding client-side adapter file setup in prebid.js.

These changes are mainly to support video-type requests through this type of workflow.

### CODE REVIEW

**Comment by bretg** [comment]
While it's ok for s2s to not have a client-side adapter, vendors and pubs should realize it's potentially a disadvantage to not include the client in the PBJS package. Even if the main auction logic happens on the server side, there may be client-side functions -- currently there's usersync, but I believe there was another feature discussed.

**Comment by jsnellbaker** [comment]
@bretg Thanks for the feedback.  I'll put together a docs PR soon.

**Comment by jsnellbaker** [comment]
To reviewers - I realized there is a workflow regarding aliasing bidders that I didn't incorporate when putting these changes together.  Please wait from reviewing until I compile these additional changes and update the PR.

**Comment by jsnellbaker** [comment]
I pushed in the updates to support aliasing bidders.

**Comment by jsnellbaker** [comment]
Created docs PR for these updates: https://github.com/prebid/prebid.github.io/pull/834

**Comment by smithaammassamveettil** [comment]
We are waiting for this PR merge to start our integration testing for S2S.
Can you please let us know what is the ETA for merging this PR?

File: modules/prebidServerBidAdapter.js [hunk 1]
```diff
@@ -734,7 +735,13 @@ export function PrebidServer() {
       utils.logError('error parsing response: ', result.status);
     }
 
-    done();
+    const videoBid = bids.some(bidResponse => bidResponse.bid.mediaType === 'video');
+    const cacheEnabled = config.getConfig('cache.url');
+
+    // video bids with cache enabled need to be cached first before they are considered done
```
Comment (mkendall07) [inline modules/prebidServerBidAdapter.js]: is this a separate change?

Comment (jsnellbaker) [reply modules/prebidServerBidAdapter.js]: No those changes are part of this PR.  They are needed to ensure pbs video bids that are using prebid cache are properly used in the auction.  Without this change; the adapter calls the `done()` before the caching process has finished which ends the auction prematurely.

**Comment by smithaammassamveettil** [comment]
*@jsnellbaker <https://github.com/jsnellbaker> *
What is the status of this PR? One of our publisher is ready to start
testing our brightroll S2S adapter. Can they do that with existing
Prebid.js they have already? Or is this PR a blocker for testing? If so,
when can I expect these changes? Please advise.

**Comment by mkendall07** [comment]
@smithaammassamveettil 
This still needs review - hopefully should be in the next release. @jsnellbaker need to resolve 1 conflict.

**Comment by jsnellbaker** [comment]
@mkendall07 I resolved the conflicts.  Please take a look over again when you have the chance.  Thanks.

**Pull request merged**